frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts

```diff
@@ -0,0 +1,108 @@
+import { execSync } from 'child_process';
+import * as _ from 'lodash';
+import { browser, ExpectedConditions as until } from 'protractor';
+import { appHost } from '@console/internal-integration-tests/protractor.conf';
+import { click } from '@console/shared/src/test-utils/utils';
+import { isNodeReady } from '@console/shared/src/selectors/node';
+import {
+  addCapacityLbl,
+  addCapacityBtn,
+  ocsOp,
+  storageClusterView,
+  storageClusterRow,
+} from '../views/add-capacity.view';
+
+const NAMESPACE = 'openshift-storage';
+const KIND = 'storagecluster';
+const EXPANDWAIT = 150000;
+const storageCluster = JSON.parse(
+  execSync(`kubectl get -o json -n ${NAMESPACE} ${KIND}`).toString(),
+);
+
+let clusterJSON;
+let previousCnt;
+let previousPods;
+let updatedClusterJSON;
+let updatedCnt;
+let updatedPods;
+
+describe('Check add capacity functionality for ocs service', () => {
+  beforeAll(async () => {
+    // check whether the cluster is present or not, and check for its ready state
+    // if not present exit and console that no cluster is present, will be send in next PR
+
+    const { name } = storageCluster.items[0].metadata;
+    clusterJSON = JSON.parse(
+      execSync(`kubectl get -o json -n ${NAMESPACE} ${KIND} ${name}`).toString(),
+    );
+    previousCnt = _.get(clusterJSON, 'spec.storageDeviceSets[0].count', undefined);
+    const uid = _.get(clusterJSON, 'metadata.uid', undefined).toString();
+    previousPods = JSON.parse(execSync(`kubectl get pods -n ${NAMESPACE} -o json`).toString());
+
+    await browser.get(
+      `${appHost}/k8s/ns/openshift-storage/operators.coreos.com~v1alpha1~ClusterServiceVersion`,
+    );
+
+    await click(ocsOp);
+    await click(storageClusterView);
+
+    await browser.wait(until.presenceOf(storageClusterRow(uid)));
+    const kebabMenu = storageClusterRow(uid).$('button[data-test-id="kebab-button"]');
+
+    await click(kebabMenu);
+    await click(addCapacityLbl);
+    await click(addCapacityBtn);
```

Comment:
when `addCapacityBtn` is clicked, if there is any error the modal won't close; instead it shows an error on the modal.
This means expansion failed validation and cluster won't be updated. Shouldn't we handle that here?

Reply:
Tier 1 test case does not include this.

Reply:
It's not about Tier 1. It could possibly crash your entire test if the modal does not transition as expected. If you think it's safe, I am ok with it.

Reply:
In this flow, we are by default value accepting `2Tib` , so this situation is very unlikely. Hence, will address this in follow up PR.